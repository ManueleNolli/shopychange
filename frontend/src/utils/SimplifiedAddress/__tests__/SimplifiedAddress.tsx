import React from 'react'
import {
  simplifiedAddress,
  renderAddress,
  renderAddressWithTokenId,
} from '../SimplifiedAddress'
import { render } from '@testing-library/react'
import { Link } from 'react-router-dom'

jest.mock('react-router-dom', () => ({
  Link: jest.fn(),
}))

describe('SimplifiedAddress', () => {
  test('simplifiedAddress', () => {
    expect(
      simplifiedAddress('0x1234567890123456789012345678901234567890')
    ).toEqual('0x1234...7890')
  })

  test('renderAddress', () => {
    ;(Link as unknown as jest.Mock).mockImplementation(
      ({ children }) => children
    )

    const tree = render(
      renderAddress('0x1234567890123456789012345678901234567890', '/nft/1')
    )

    expect(tree.container).toMatchSnapshot()
  })

  test('renderAddressWithTokenId', () => {
    ;(Link as unknown as jest.Mock).mockImplementation(
      ({ children }) => children
    )
    const address = '0x1234567890123456789012345678901234567890'
    const tokenId = 123
    const navigate = '/path/to/navigate'

    const tree = render(renderAddressWithTokenId(address, tokenId, navigate))

    expect(tree.container).toMatchSnapshot()
  })
})
