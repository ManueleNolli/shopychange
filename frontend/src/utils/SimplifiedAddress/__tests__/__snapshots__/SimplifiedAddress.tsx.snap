// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplifiedAddress renderAddress 1`] = `
<div>
  <p
    class="chakra-text css-0"
  >
    0x1234...7890
  </p>
</div>
`;

exports[`SimplifiedAddress renderAddressWithTokenId 1`] = `
<div>
  <p
    class="chakra-text css-0"
  >
    0x1234...7890
    /
    123
  </p>
</div>
`;
